core/common/src/main/java/alluxio/master/metastore/rocks/RocksExclusiveLockHandle.java

```diff
@@ -0,0 +1,46 @@
+package alluxio.master.metastore.rocks;
+
+import com.google.common.base.Preconditions;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * This is a handle used to close the write lock(exclusive lock) on RocksStore.
+ * The exclusive lock is acquired when ref count is zero, and the StopServingFlag ensures
+ * no new r/w will come in, so the ref count will stay zero throughout the period.
+ */
+public class RocksExclusiveLockHandle implements AutoCloseable {
+  /*
+   * true: the RocksDB is temporarily closed and can serve after the lock is released
+   * false: the RocksDB is closed permanently because the master is shutting down
+   */
+  final boolean mIsRestart;
+  final AtomicBoolean mStopServingFlag;
+  final LongAdder mRefCount;
+
+  /**
+   * The constructor.
+   *
+   * @param isRestart if true, reset the StopServingFlag so the RocksDB accepts new operations
+   * @param stopServingFlag the flag on RocksStore
+   * @param refCount the ref count on RocksStore
+   */
+  public RocksExclusiveLockHandle(
+      boolean isRestart, AtomicBoolean stopServingFlag, LongAdder refCount) {
+    mIsRestart = isRestart;
+    mStopServingFlag = stopServingFlag;
+    mRefCount = refCount;
+  }
+
+  @Override
+  public void close() {
+    Preconditions.checkState(mRefCount.sum() == 0,
```

Comment:
I feel this check may be a bit too strict. A bug (an unfound RocksDB access which does not respect the flag) will likely crash the master process here, whereas without this check it's only a low chance to segfault.

I think the use of this check is, like the leak detector, to detect all illegal accesses in test time (especially UT). So it's ideally a property which we enable in tests, and disable in production (reduce to only log a warning).

Reply:
In prod environment we want to be tolerant and lenient to small data races. In other words if the sum is not zeroed after a timeout, we print a warning and proceed (instead of crashing).

But in tests we want to be very strict, meaning if there's a race we want the process to crash in a super obvious way so we find the issue and fix it. 

That difference suggests we want have a config, like `alluxio.leak.detector.level`. We use `alluxio.leak.detector.level=DISABLED` in real deployment but use `alluxio.leak.detector.level=PARANIOD` in tests. `TEST_MODE` happens to be such a config i think I'm going to use here.